flow/connectors/bigquery/merge_statement_generator.go

```diff
@@ -181,6 +181,20 @@ func (m *mergeStmtGenerator) generateMergeStmt() string {
 		pkeySelectSQL, insertColumnsSQL, insertValuesSQL, updateStringToastCols, deletePart)
 }
 
+func (m *mergeStmtGenerator) generateMergeStmts() []string {
+	// TODO (kaushik): This is so that the statement size for individual merge statements
+	// doesn't exceed the limit. We should make this configurable.
+	const batchSize = 8
+	partitions := utils.ArrayPartition(m.unchangedToastColumns, batchSize)
```

Comment:
This could instead be an iterator,
```go
type ChunkIter[T] struct {
  slice: T[]
  index: int
  chunkSize: int
}
```
then
```go
for chunk in chunks.Next() {
  mergeStmts = append(mergeStmts, g.generateMergeStmt(chunk))
}
```

Reply:
From current experience it seems to be 10-30, doesn't seem worth it to implement an iterator for this